test/parallel/test-worker-abort.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+const common = require('../common');
+const { Worker, isMainThread } = require('worker_threads');
+
+if (isMainThread) {
+  const ac = new AbortController();
+  const w = new Worker(__filename, { signal: ac.signal });
+  w.on('online', common.mustCall(() => {
```

Comment:
When passing `AbortSignal.abort` I would expect the worker to never become online